src/dispatchers/dispatcher.ts

```diff
@@ -178,26 +177,23 @@ export class DispatcherConnection {
     try {
       const validated = this._validateParams(dispatcher._type, method, params);
       const result = await (dispatcher as any)[method](validated, this._validateMetadata(metadata));
-      this.onmessage({ id, result: this._replaceDispatchersWithGuids(result, true) });
+      this.onmessage({ id, result: this._replaceDispatchersWithGuids(result) });
     } catch (e) {
       this.onmessage({ id, error: serializeError(e) });
     }
   }
 
-  private _replaceDispatchersWithGuids(payload: any, allowDispatchers: boolean): any {
+  private _replaceDispatchersWithGuids(payload: any): any {
     if (!payload)
       return payload;
-    if (payload instanceof Dispatcher) {
-      if (!allowDispatchers)
-        throw new Error(`Channels are not allowed in the scope's initialzier`);
```

Comment:
This particular check was a result of the following problem:

BrowserContext had `pages` in the initializer, and since we send channels from the initializer before the object, these pages were created in the parent scope (the Browser), rather than BrowserContext's scope. This lead to leaks where pages were not cleaned up upon BrowserContext's dispose. I am afraid this could happen again.

Reply:
This looks like an edge case, let's add a test instead?